client/src/main/java/io/pravega/client/stream/notifications/ReaderGroupEventListener.java

```diff
@@ -19,9 +19,17 @@
 public interface ReaderGroupEventListener {
 
     /**
-     * Get a scale event notifier for a given readergroup. A scale event notifier is triggered when there are more
-     * number of segments when compared to the number of readers or vice versa. Listeners can be registered which will
-     * notify in case of a scale event.
+     * Get a scale event notifier for a given reader group.
+     * <br>
+     * A scale event notifier is triggered when the total number of segments managed by the ReaderGroup changes. During
```

Comment:
This notifier isn't really about scale events any longer. It is about changes to the number of segments, so we should consider updating the names of classes and descriptions accordingly. 

As an example of what I'm arguing, say that I have three segments S1, S2, S3 that at time T merges into S4. This is one scale event, but there could be multiple changes to the number of segments. Say S1 hits the end first and adds S4 to future segments. The number of segments changes from 3 to 2, so one event is generated. Now when S2 hits the end, the number changes from 2 to 1, and another notification is generated.

Reply:
Updated the names of classes and description e.g: ScaleEventNotifier to SegmentEventNotifier.